hoomd/md/CylinderManifold.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2009-2020 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+
+// Maintainer: pschoenhoefer
+
+#include "CylinderManifold.h"
+
+namespace py = pybind11;
+
+using namespace std;
+
+/*! \file CylinderManifold.cc
+    \brief Contains code for the CylinderManifold class
+*/
+
+/*!
+    \param r radius of the sphere
+    \param P position of the sphere
+*/
+CylinderManifold::CylinderManifold(std::shared_ptr<SystemDefinition> sysdef,
+                               Scalar r,
+                               Scalar3 P)
+  : Manifold(sysdef), m_r(r) 
+       {
+    m_exec_conf->msg->notice(5) << "Constructing CylinderManifold" << endl;
```

Comment:
I don't think the `Manifold*` classes need a `SystemDefinition` or an `ExecutionConfiguration`. They don't need the notice messages and they don't operate directly on particles.

You could modify the `validate` method to take the `BoxDim` as an argument and raise a `RuntimeError` with the error message.